### PR TITLE
Make distance dispatch to parent for embedded manifold

### PR DESCRIPTION
**Pull request by sethaxen**
Fixes part of https://github.com/JuliaManifolds/Manifolds.jl/issues/338.

### CODE REVIEW

File: src/EmbeddedManifold.jl [hunk 1]
```diff
@@ -252,6 +252,20 @@ function decorator_transparent_dispatch(
...
+function decorator_transparent_dispatch(
+    ::typeof(distance),
+    ::AbstractEmbeddedManifold,
+    args...,
+)
+    return Val(:parent)
+end
+function decorator_transparent_dispatch(
+    ::typeof(distance),
+    ::AbstractEmbeddedManifold{𝔽,<:AbstractIsometricEmbeddingType},
```
Comment (mateuszbaran) [inline src/EmbeddedManifold.jl:264]: I think this should be restricted to `TransparentIsometricEmbedding`.
```suggestion
    ::AbstractEmbeddedManifold{𝔽,TransparentIsometricEmbedding},
```

Comment (sethaxen) [reply src/EmbeddedManifold.jl]: That's interesting. `exp` and `log` both have `TransparentIsometricEmbedding` here, while `norm` has `AbstractIsometricEmbeddingType`. All three depend on the metric though. Why the different overloads for these three?

Comment (mateuszbaran) [reply src/EmbeddedManifold.jl]: Actually, I have no idea why `norm` is `transparent` for `AbstractIsometricEmbedding`. As far as I understand it, `norm` should be `parent` for non-transparent embedding and `transparent` for transparent embedding. Maybe @kellertuer knows why it's different?

Comment (mateuszbaran) [reply src/EmbeddedManifold.jl]: Ah, right, I was wrong. It's correct. `AbstractIsometricEmbedding` means that `inner` and `norm` can be transparently forwarded while `exp` and `log` can only be directly forwarded for `TransparentIsometricEmbedding`. That's correct. That's basically what these embedding types mean.

Comment (mateuszbaran) [reply src/EmbeddedManifold.jl]: Still, `norm` should probably be `:parent` for `AbstractEmbeddedManifold`?

Comment (mateuszbaran) [reply src/EmbeddedManifold.jl:264]: @sethaxen Why did you change it back to `<:AbstractIsometricEmbeddingType`?

Comment (sethaxen) [reply src/EmbeddedManifold.jl]: Whoops, accident.

**Review by mateuszbaran — APPROVED** [review]

**Review by kellertuer — APPROVED** [review]

**Pull request merged**